compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -443,14 +452,32 @@ class ProtobufGenerator(val params: GeneratorParams) extends DescriptorPimps {
         fp.add(s"if (${toBaseType(field)(fieldNameSymbol)} != ${defaultValueForGet(field, true)}) { __size += ${sizeExpressionForSingleField(field, toBaseType(field)(fieldNameSymbol))} }")
       }
     } else if (field.isOptional) {
-      fp.add(s"if ($fieldNameSymbol.isDefined) { __size += ${sizeExpressionForSingleField(field, toBaseType(field)(fieldNameSymbol + ".get"))} }")
+      if (field.customSingleScalaTypeName.isDefined) {
+        fp.add(
+          s"if ($fieldNameSymbol.isDefined) {",
+          s"  val __value = ${toBaseType(field)(fieldNameSymbol + ".get")}",
```

Comment:
Would it make sense to use this code for both cases (and then remove the `if`?) This would reduce duplication.